sympy/polys/eigentools.py

```diff
@@ -0,0 +1,108 @@
+from sympy.polys import Poly
+from sympy.polys.domainmatrix import DDM, ddm_irref, ddm_isub
+from sympy.polys.factortools import dup_factor_list
+from sympy.polys.agca.extensions import FiniteExtension, ExtensionElement
+from sympy.core.symbol import Dummy
+from sympy.polys.rootoftools import CRootOf
+from sympy.polys.polyroots import roots
+
+
+def ddm_nullspace(M):
+    rows, cols = M.shape
+    domain = M.domain
+
+    pivots = ddm_irref(M)
+    rref = M
+
+    basis = []
+    for i in range(cols):
+        if i in pivots:
+            continue
+        vec = [domain.one if i == j else domain.zero for j in range(cols)]
+        for ii, jj in enumerate(pivots):
+            vec[jj] -= rref[ii][i]
+        basis.append(vec)
+
+    return basis
+
+
+def ddm_eigenvects(A, l=Dummy('lambda')):
+    charpoly = A.charpoly()
+    rows, cols = A.shape
+    domain = A.domain
+    _, factors = dup_factor_list(charpoly, domain)
+
+    eigenvects = []
```

Comment:
I haven't made into a full picture how to represent eigenvectors in most generic way, and in least fragile way.
There should is an obvious reason to distinguish the classes where eigenvalues belong to the coefficient domain and eigenvectors belong to the algebraic extension, which is different from what was done for Matrix.
So before I figure out the best representation, it's not implemented under DomainMatrix method.